src/Workspaces/Core/Portable/Diagnostics/DiagnosticDataSerializer.cs

```diff
@@ -110,21 +98,12 @@ public void WriteTo(ObjectWriter writer, ImmutableArray<DiagnosticData> items, C
                 writer.WriteBoolean(item.IsSuppressed);
                 writer.WriteInt32(item.WarningLevel);
 
-                if (item.HasTextSpan)
-                {
-                    // document state
-                    writer.WriteInt32(item.TextSpan.Start);
-                    writer.WriteInt32(item.TextSpan.Length);
-                }
-                else
-                {
-                    // project state
-                    writer.WriteInt32(0);
-                    writer.WriteInt32(0);
-                }
+                // unused
+                writer.WriteInt32(0);
+                writer.WriteInt32(0);
```

Comment:
Can we not just remove these?

Reply:
I'm not sure what the protocol versioning requirements are here. If we removed this we would need to bump the format version. I did not want to do that in this PR.